game-core/src/main/java/games/strategy/engine/chat/HeadlessChat.java

```diff
@@ -17,7 +17,7 @@
   // roughly 1000 chat messages
   private static final int MAX_LENGTH = 1000 * 200;
   private Chat chat;
-  private StringBuilder allText = new StringBuilder();
+  private final StringBuilder allText = new StringBuilder();
   private final ChatFloodControl floodControl = new ChatFloodControl();
 
   public HeadlessChat(
```

Comment:
We may want to investigate where/how messages get into HeadlessChat and if they should be truncated at the chat controller level. I think the problem is that we use a 'broadcaster' for chat instead of a remote. It might be better for chat to be sent by getting a remote, that would only call a method on chat controller. Controller would then use a broadcaster to relay the message. In that way, instead of clients blasting out messages through the controller to all other clients, we could have some centralized code. What is otherwise happening is a message that is too long is being broadcast and then truncated server side.

I'm thining about this also in the context of a headlessChat, we likely should not need any truncation code here at all, either becasue the controller has done this work for us, or, because it's a bot, it's just not needed.

Reply:
I believe this is supposed to be a buffer for chat messages. Once the storage gets full, the code deletes the oldest half of all messages and therefore regains half of the space again.
There's probably a better way to achieve that using `EvictingQueue` or something, but this is probably outside of the scope of this PR.

Reply:
Okay. Kinda convoluted IMO, the 'allText' is only here as a buffer for when a moderator requests the chat log of a bot. I think the the coding here lost vision of its intent/purpose